setup/mscs/msc2946/msc2946.go

```diff
@@ -45,6 +45,9 @@ const (
 	ConstCreateEventContentValueSpace = "m.space"
 	ConstSpaceChildEventType          = "m.space.child"
 	ConstSpaceParentEventType         = "m.space.parent"
+	ConstJoinRulePublic               = "public"
```

Comment:
We have constants defined in gomatrixserverlib already: `gomatrixserverlib.Public`, `gomatrixserverlib.Knock` and `gomatrixserverlib.Restricted`. Please use those instead of defining additional ones.

Reply:
Awesome suggestion! I made those changes, thanks for the heads up.